test/grouping.jl

```diff
@@ -22,4 +23,31 @@ d = DataFrame(n = 1:20, x = [3, 3, 3, 3, 1, 1, 1, 2, 1, 1, 2, 1, 1, 2, 2, 2, 3,
 g = groupby(d, :x, sort=true)
 @test @based_on(g, nsum = sum(:n))[:nsum] == [99, 84, 27]
 
+# Transform tests
+d = DataFrame(a = [1,1,1,2,2,3,3,1], 
+              b = Any[1,2,3,missing,missing,6.0,5.0,4], 
+              c = CategoricalArray([1,2,3,1,2,3,1,2]))
+g = groupby(d, :a, sort = false)
+## Scalar output 
+# Type promotion Int -> Float
+@test isequal(@transform(g, t = :b[1])[:t], 
```

Comment:
`isequal` doesn't check types. Writing `1.0` and `6.0` is even misleading since it's strictly equivalent to `1` and `6`. You need either to store the result of the operation and check its type manually. Using `.===` as you did below is a clever trick, but it's not enough to ensure that the element type is e.g. `Union{Float64,Missing}` rather than `Any`.

Reply:
You probably don't need `.===` if you use `typeof`, at least not if all numbers are expected to be of the same type (which is the case here if I'm not mistaken).